src/app/Media/IMediaSession.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace SIPSorcery.SIP.App
+{
+    /// <summary>
+    /// Offering and Answering SDP messages so that it can be
+    /// signaled to the other party using the SIPUserAgent.
+    /// 
+    /// The implementing class is responsible for ensuring that the client
+    /// can send media to the other party including creating and managing
+    /// the RTP streams and processing the audio and video.
+    /// </summary>
+    public interface IMediaSession
+    {
+        Task<string> CreateOffer(IPAddress destinationAddress = null);
+        Task OfferAnswered(string remoteSDP);
+
+        Task<string> AnswerOffer(string remoteSDP);
+        Task<string> RemoteReInvite(string remoteSDP);
+
```

Comment:
Do these methods need to be async? There's no network communications or DNS involved. Mostly the methods are only doing string parsing with one OS call to interrogate the in memory routing table which should be very fast.

Reply:
I left it non-async until I realized that my personal implementation that uses IceLink works with Tasks.

Reply:
Ah ok, doesn't hurt I guess. It would be worth adding a one line comment mentioning the reason to stop it being inadvertently changed in the future.